x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/AbstractTransportGetResourcesAction.java

```diff
@@ -70,7 +71,10 @@ protected AbstractTransportGetResourcesAction(String actionName, TransportServic
     protected void searchResources(AbstractGetResourcesRequest request, ActionListener<QueryPage<Resource>> listener) {
         String[] tokens = Strings.tokenizeToStringArray(request.getResourceId(), ",");
         SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
-            .sort(request.getResourceIdField())
+            .sort(SortBuilders.fieldSort(request.getResourceIdField())
```

Comment:
Is it possible to a unit test for that to prevent regression?
Bonus question: do we have something like in-memory search API for unit testing? Or does this kind of test has to be in integration test framework?

Reply:
I looked into adding a test. There's no much value adding a unit test as I'd just check the code itself, not the effect. I tried adding a YML test but I could not reproduce the bug. I think this was because the index was already created by the filter_crud.yml tests. I tried deleting the `.ml-meta` index but it didn't seem to work. I'll give it another shot.

Reply:
Ah, I was forgetting the problem does not arise when the index is not there, but when the index is there with no mapping for `filter_id`. Test added! Thanks for the prompt!